src/util/check-github-app-type.ts

```diff
@@ -19,3 +28,23 @@ export const setGitHubBaseUrl = (gitHubBaseUrl: string | undefined): string => {
 export const setAcceptHeader = (gitHubBaseUrl: string | undefined): string => {
 	return gitHubBaseUrl ? "application/vnd.github.machine-man-preview+json" : "application/vnd.github.v3+json";
 }
+
+/**
+ * Factory function to create a GitHub client that authenticates as the installation of our GitHub app.
+ */
+export async function createInstallationClient(githubInstallationId: number, jiraHost: string, logger: Logger): Promise<GitHubInstallationClient> {
+	const githubBaseUrl = await getGitHubBaseUrl(jiraHost);
+	return await booleanFlag(BooleanFlags.GHE_SERVER_AUTH_AND_CONNECT_FLOW, false, jiraHost)
+		? new GitHubInstallationClient(getCloudInstallationId(githubInstallationId, githubBaseUrl), logger, githubBaseUrl)
+		: new GitHubInstallationClient(getCloudInstallationId(githubInstallationId), logger);
+}
+
+/**
+ * Factory function to create a GitHub client that authenticates as the user (with a user access token).
+ */
+export async function createUserClient(githubToken: string, jiraHost: string, logger: Logger): Promise<GitHubUserClient> {
+	const githubBaseUrl = await getGitHubBaseUrl(jiraHost);
+	return await booleanFlag(BooleanFlags.GHE_SERVER_AUTH_AND_CONNECT_FLOW, false, jiraHost)
+		? new GitHubUserClient(githubToken, logger, githubBaseUrl)
+		: new GitHubUserClient(githubToken, logger);
+}
```

Comment:
Once all occurrences of `getGitHubBaseUrl` are refactored to the new `create...` functions, that makes the feature flagging a lot simpler in the code, already. The next step, then, would be to get rid of the `setGitHubBaseUrl` and `setAcceptHeader` functions and instead pass an object into the constructor of the clients that contains the baseUrl and the headers (or functions that get them).